javascript/react/src/hooks/useWorkflowSteps.ts

```diff
@@ -3,19 +3,35 @@
  * Licensed under the MIT license. See LICENSE file in the project.
  */
 import type { Step, Workflow } from '@datashaper/workflow'
+import cloneDeep from 'lodash-es/cloneDeep.js'
 import { useEffect, useState } from 'react'
 
+import { WorkflowOrder } from './../components/ManageWorkflow.types.js'
+
 /**
  * Gets the workflow processing steps
  * @param workflow - the workflow intsance
  * @returns
  */
-export function useWorkflowSteps(workflow: Workflow): Step[] {
+export function useWorkflowSteps(
+	workflow: Workflow,
+	order: WorkflowOrder,
+): Step[] {
 	const [steps, setSteps] = useState<Step[]>(workflow.steps)
 	// listen for workflow changes and update the steps
 	useEffect(() => {
-		setSteps(workflow.steps)
-		workflow.onChange(() => setSteps(workflow.steps))
-	}, [workflow, setSteps])
+		setSteps(orderSteps(workflow.steps, order))
+		workflow.onChange(() => {
+			setSteps(orderSteps(workflow.steps, order))
+		})
+	}, [workflow, order, setSteps])
+	return steps
+}
+
+function orderSteps(_steps: Step[], order: WorkflowOrder): Step[] {
+	let steps = cloneDeep(_steps)
```

Comment:
Do you need `cloneDeep` or can you just do something like `[..._steps]`?

Reply:
I guess you could go one step further too:
```
const steps = [..._steps]
return order === WorkflowOrder.FirstOnTop ? steps : steps.reverse()
```

Reply:
Yeahh that works, thanks!